packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -255,15 +261,42 @@ class AppBar extends StatefulWidget implements PreferredSizeWidget {
 
   /// The primary widget displayed in the app bar.
   ///
-  /// Typically a [Text] widget containing a description of the current contents
-  /// of the app.
+  /// Typically a [Text] widget that contains a description of the current
+  /// contents of the app.
+  ///
+  /// Becomes the middle component of the [NavigationToolBar] built by this widget.
+  /// The [title]'s width is constrained to fit within the toolbar's [leading]
+  /// and [actions] widgets. Its height is _not_ constrained. The [title]
+  /// is vertically centered and clipped to fit within the toolbar, whose
+  /// height is [kToolbarHeight]. Typically this isn't noticeable because
+  /// a simple [Text] [title] will fit within the toolbar by default.
+  /// On the other hand, it is noticeable when a title widget with an intrinsic
```

Comment:
For devs less familiar with the term intrinsic height, maybe add a sentence to describe a common situation where the `title` widget ends up being clipped. In the issue, it seems like images tend to run into this scenario, so, something like: `... For example, when the Image used in the [title] has a height that exceeds [kToolbarHeight], it will be clipped. In cases like this...`

Reply:
Updated the text per your suggestion.